pkg/kubelet/kubelet_pods_test.go

```diff
@@ -3284,6 +3284,7 @@ func TestGenerateAPIPodStatusHostNetworkPodIPs(t *testing.T) {
 			criPodIPs: []string{"192.168.0.1"},
 			podIPs: []v1.PodIP{
 				{IP: "192.168.0.1"},
+				{IP: "fd01::1234"},
```

Comment:
Interesting corner case from DanWinships test cases
```
			name: "CRI PodIPs override NodeAddresses",
			nodeAddresses: []v1.NodeAddress{
				{Type: v1.NodeInternalIP, Address: "10.0.0.1"},
				{Type: v1.NodeInternalIP, Address: "fd01::1234"},
			},
			criPodIPs: []string{"192.168.0.1"},
			podIPs: []v1.PodIP{
				{IP: "192.168.0.1"},
				{IP: "fd01::1234"},
```

CRIO and containerd doesn't send IPs for host network pods
https://github.com/cri-o/cri-o/blob/8d4d158935929800c4300b82eb4b5a83ded400f4/server/sandbox_network.go#L34-L36

https://github.com/containerd/containerd/blob/4921fb6b63e5476e54fb7e2cf93c3b505f1c09cb/pkg/cri/store/sandbox/sandbox.go#L40-L41

but they can.
I think that the discussion here is if override means removing IPs or just updating the ones that it sets leaving the others as is

Reply:
I think override means "ignore CRI".

Do we think it is EVER correct for CRI to return host-net IPs?

Reply:
> Do we think it is EVER correct for CRI to return host-net IPs?

I think we all agree that is a bad idea, but it seems that it is the way it was https://github.com/kubernetes/kubernetes/pull/106715#issuecomment-985943968

Reply:
@danwinship what do you think?

Reply:
If *nothing* else we should specify what Kubernetes expects host-network pod IPs to be (order, number, type). I don't like leaving that an undefined wild-west.

And I think we should enforce that expectation. If there are corner cases that are currently possible that we break, I think a KEP should get written to implement that corner-case correctly.

Reply:
Just for reference, the CRI IPs are obtained in kuberuntime in `GetPodStatus()`

https://github.com/kubernetes/kubernetes/blob/571e067cddfbb0c2a3092ab9d3be3074043e3810/pkg/kubelet/kuberuntime/kuberuntime_manager.go#L1064

and it seems they are stored in the PLEG cache

https://github.com/kubernetes/kubernetes/blob/571e067cddfbb0c2a3092ab9d3be3074043e3810/pkg/kubelet/pleg/generic.go#L407-L419

This cache is used to `syncPod()` from the PodWorker

https://github.com/kubernetes/kubernetes/blob/571e067cddfbb0c2a3092ab9d3be3074043e3810/pkg/kubelet/pod_workers.go#L904

and `generateAPIPodStatus()`

https://github.com/kubernetes/kubernetes/blob/571e067cddfbb0c2a3092ab9d3be3074043e3810/pkg/kubelet/kubelet.go#L1554

Reply:
it seems there is no way to differentiate between:
- the CRI setting the IPs from CRI
- CRI NOT returning IPs, but Pod status IPs updated incrementally

Reply:
I've updated the commit with the CRI API change

Reply:
Yes, this sounds good from the CRI API perspective :+1: